python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -1,1140 +1,1164 @@
-      [](const std::vector<std::string> &hashes) {
-        for (auto iter = cudaqStateStorage->cbegin();
-             iter != cudaqStateStorage->cend();) {
-          if (std::find(hashes.begin(), hashes.end(), iter->first) !=
-              hashes.end()) {
-            cudaqStateStorage->erase(iter++);
-            continue;
-          }
-          iter++;
-        }
-      },
-      "Remove our pointers to the cudaq states.");
-
-  mod.def(
-      "mergeExternalMLIR",
-      [](MlirModule modA, const std::string &modBStr) {
-        auto ctx = unwrap(modA).getContext();
-        auto moduleB = parseSourceString<ModuleOp>(modBStr, ctx);
-        auto moduleA = unwrap(modA).clone();
-        moduleB->walk([&moduleA](func::FuncOp op) {
-          if (!moduleA.lookupSymbol<func::FuncOp>(op.getName()))
-            moduleA.push_back(op.clone());
-          return WalkResult::advance();
-        });
-        return wrap(moduleA);
-      },
-      "Merge the two Modules into a single Module.");
-
-  mod.def(
-      "synthPyCallable",
-      [](MlirModule modA, const std::vector<std::string> &funcNames) {
-        auto m = unwrap(modA);
-        auto context = m.getContext();
-        PassManager pm(context);
-        pm.addNestedPass<func::FuncOp>(
-            cudaq::opt::createPySynthCallableBlockArgs(
-                SmallVector<StringRef>(funcNames.begin(), funcNames.end()),
-                true));
-        if (failed(pm.run(m)))
-          throw std::runtime_error(
-              "cudaq::jit failed to remove callable block arguments.");
-
-        // fix up the mangled name map
-        DictionaryAttr attr;
-        m.walk([&](func::FuncOp op) {
-          if (op->hasAttrOfType<UnitAttr>("cudaq-entrypoint")) {
-            auto strAttr = StringAttr::get(
-                context, op.getName().str() + "_PyKernelEntryPointRewrite");
-            attr = DictionaryAttr::get(
-                context, {NamedAttribute(StringAttr::get(context, op.getName()),
-                                         strAttr)});
-            return WalkResult::interrupt();
-          }
-          return WalkResult::advance();
-        });
-        if (attr)
-          m->setAttr("quake.mangled_name_map", attr);
-      },
-      "Synthesize away the callable block argument from the entrypoint in "
-      "`modA` with the `FuncOp` of given name.");
-
-  mod.def(
-      "jitAndGetFunctionPointer",
-      [](MlirModule mod, const std::string &funcName) {
-        auto jit = jitKernel(funcName, mod, {});
-        auto funcPtr = jit->lookup(funcName);
-        if (!funcPtr) {
-          throw std::runtime_error(
-              "cudaq::builder failed to get kernelReg function.");
-        }
-
-        return py::capsule(*funcPtr);
-      },
-      "JIT compile and return the C function pointer for the FuncOp of given "
-      "name.");
-}
-} // namespace cudaq
+/*******************************************************************************

```

Comment:
Why is every single line of this file changed? How are we supposed to review this?